Switch to PEP 420 native namespaces

erral opened the pull request.

mauritsvanrees reviewed (commented):
I made a few minor changes.  For example we do need to call `find_namespace_packages` in `setup.py`.

Why did you change the minimum required Python to 3.11?  Did you hear a rumour somewhere?  I am not planning to change it, it should just be 3.10.  I have updated the branch.

In `.meta.toml` we should add some lines for good measure:

```
[tox]
constraints_file = "https://dist.plone.org/release/6.2-ev/constraints.txt"
```

But then we would need to call plone/meta on it, which makes too many changes.

erral commented:
> I made a few minor changes. For example we do need to call `find_namespace_packages` in `setup.py`.
> 
> Why did you change the minimum required Python to 3.11? Did you hear a rumour somewhere? I am not planning to change it, it should just be 3.10. I have updated the branch.

No, sorry, I wanted to update it to the actual version, and thought that it was 3.10.

mauritsvanrees reviewed (commented):
I have rebased on master, squashed the commits, added 6.2-dev as constraints, and have force pushed.

LGTM now, but we should not merge yet until the moment is there that we really start moving all plone packages to native namespaces.

erral commented:
> I have rebased on master, squashed the commits, added 6.2-dev as constraints, and have force pushed.
> 
> LGTM now, but we should not merge yet until the moment is there that we really start moving all plone packages to native namespaces.

Perfect, that's why this is a draft PR

erral commented:
Superceded by #567
